Meadowlark-js/backends/meadowlark-postgresql-backend/src/repository/Delete.ts

```diff
@@ -27,43 +31,32 @@ export async function deleteDocumentByDocumentUuid(
   let meadowlarkId: MeadowlarkId = '' as MeadowlarkId;
 
   try {
-    await client.query('BEGIN');
+    await beginTransaction(client);
     // Find the alias meadowlarkIds for the document to be deleted
-    const documentAliasIdsResult: QueryResult = await client.query(
-      findAliasMeadowlarkIdsForDocumentByDocumentUuidSql(documentUuid),
+    const documentAliasIdsResult: MeadowlarkAlias[] = await findAliasMeadowlarkIdsForDocumentByDocumentUuid(
+      client,
+      documentUuid,
     );
     // Each row contains documentUuid and corresponding meadowlarkId (meadowlark_id),
     // we just need the first row to return the meadowlark_id
     meadowlarkId =
-      documentAliasIdsResult?.rowCount != null && documentAliasIdsResult.rowCount > 0
-        ? documentAliasIdsResult.rows[0].meadowlark_id
-        : ('' as MeadowlarkId);
+      (documentAliasIdsResult?.length ?? 0) > 0 ? documentAliasIdsResult[0].meadowlark_id : ('' as MeadowlarkId);
     if (validateNoReferencesToDocument) {
       // All documents have alias meadowlarkIds. If no alias meadowlarkIds were found, the document doesn't exist
       if (meadowlarkId === '') {
-        await client.query('ROLLBACK');
+        await rollbackTransaction(client);
         Logger.debug(`${moduleName}.deleteDocumentByDocumentUuid: DocumentUuid ${documentUuid} does not exist`, traceId);
         deleteResult = { response: 'DELETE_FAILURE_NOT_EXISTS' };
         return deleteResult;
       }
 
       // Extract from the query result
-      const documentAliasMeadowlarkIds: MeadowlarkId[] = documentAliasIdsResult.rows.map((ref) => ref.alias_meadowlark_id);
+      const documentAliasMeadowlarkIds: MeadowlarkId[] = documentAliasIdsResult.map((ref) => ref.alias_meadowlark_id);
 
       // Find any documents that reference this document, either it's own meadowlarkId or an alias
-      const referenceResult: QueryResult<any> = await client.query(
-        findReferencingMeadowlarkIdsSql(documentAliasMeadowlarkIds),
-      );
-
-      if (referenceResult.rows == null) {
-        await client.query('ROLLBACK');
```

Comment:
Can we really remove this? Or does it need to go into `findReferencingMeadowlarkIdsSql` and throw an exception with the error message?

Reply:
If we move that to the helper, should we have validation for all queries? Because at this moment the nulls are validated to return the empty object, but it could be that for other cases the client returns null the result.

Reply:
Got it. It's fine as is.